frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/common/PortletUtils.java

```diff
@@ -363,4 +368,50 @@ public static VerificationState getVerificationState(VulnerabilityDTO vul){
                                 .findFirst()
                                 .orElse("")));
     }
+
+    public static Map<String, Integer> getServeListMapFromRequest(ResourceRequest request) {
+        HttpServletRequest httpReq = PortalUtil.getOriginalServletRequest(PortalUtil.getHttpServletRequest(request));
+        Map<String, Integer> serveListMap = new HashMap<>();
+
+        int listDisplayAmount = 25;
+        int startIdx = 0;
+        int sortColumnIdx = 1;
+        int sortDir = 0;
+
+        String pageNo = httpReq.getParameter(DATATABLE_DISPLAY_START);
+        String pageSize = httpReq.getParameter(DATATABLE_DISPLAY_LENGTH);
+        String colIndex = httpReq.getParameter(DATATABLE_SORT_COLUMN);
+        String sortDirection = httpReq.getParameter(DATATABLE_SORT_DIRECTION);
+
+        if (pageNo != null) {
+            startIdx = Integer.parseInt(pageNo);
+            if (startIdx < 0) {
+                startIdx = 0;
+            }
+        }
+
+        if (pageSize != null) {
```

Comment:
Maybe introduce a helper function taking a `lowerBound`, `upperBound` and a `default`, which does also the null check.

You could even move the `httpReq.getParameter(` call to the helper function, and call in 381-384 only some helper four times, which handles the parsing.